### ISSUE TITLE
x/build: add a darwin longtest builder

### ISSUE THREAD

**Issue by bcmills** [golang/go#35678]
Now that we have a Windows longtest builder (#26529), I'm wondering where else we might have a coverage gap.

It would be nice to have a `darwin` builder running non-short tests too. I doubt that it will turn up many regressions for `cmd/go` in particular, but the system calls are enough different — and the installed base of `cmd/go` users on `macOS` is large enough — that I suspect it would have a net positive value.

CC @dmitshur @cagedmantis @toothrot @bradfitz

**Comment by bcmills** [comment]
(This message brought to you by `cmd/go.tooSlow`, which probably ought to just Not Be a Thing.)

https://github.com/golang/go/blob/a23f9afd9899160b525dbc10d01045d9a3f072a0/src/cmd/go/go_test.go#L49-L53

**Comment by bcmills** [comment]
CC @matloob @jayconrod

**Comment by dmitshur** [comment]
I've folded the identical #26530 into this issue. I've also retitled this one to be about darwin/amd64, since #49055 now exists for darwin/arm64.

**Comment by bcmills** [comment]
This was moved to the `Planned` project state back in November — do we have an ETA on when it will happen, or is it still in the backlog?

I'd still like to get rid of the builder check in the `tooSlow` test helper because it fails to detect builders that really are slower (like the `darwin-amd64-race` builder), such as in #58918 and #58919, and I don't want to add even more complexity to the test to try to hard-code more information about specific builders.

More clarity about when this is likely to happen would help me evaluate the tradeoffs of that change.

**Comment by bcmills** [comment]
FWIW, from my perspective the builder would be useful even if it doesn't have enough machines to run at every CL.

(Many of the reverse builders are in that state, and they're still useful for detecting deterministic failures, narrowing down bisection windows, and performing `SlowBot` runs on particularly risky changes.)

**Comment by prattmic** [comment]
This is deployed. Tests are failing because they are attempting to access the GKE proxy: https://build.golang.org/log/3f22deaeecb1b6d29789d0c3dc904a7edf34bf1c.

I don't immediately see why this is happening, or even where this value is coming from. There is no GOPROXY in the logged environment.

**Comment by bcmills** [comment]
https://cs.opensource.google/go/x/build/+/master:cmd/coordinator/buildstatus.go;l=1047;drc=f2d94d7ddd3020510adb472cc4377d4d0a1558e8

Overridden here for the non-GCE builders:
https://cs.opensource.google/go/x/build/+/master:dashboard/builders.go;l=908-912;drc=b88b71fe103eaec6c7372cf0a34f2bd7227435f6

**Comment by bcmills** [comment]
(I don't know why it isn't in the logged environment, though. 🤷‍♂️)

**Comment by prattmic** [comment]
Thanks, I found that but misread some conditions.

> (I don't know why it isn't in the logged environment, though. 🤷‍♂️)

I think this is because we are only logging the environment when running make.bash. For tests we add GOPROXY [here](https://cs.opensource.google/go/x/build/+/master:cmd/coordinator/buildstatus.go;l=1751-1753;drc=f2d94d7ddd3020510adb472cc4377d4d0a1558e8), only after running make.bash.

**Comment by prattmic** [comment]
The latest failures are in crypto/x509:

https://build.golang.org/log/8815b334c89791bc3c30410757d4c1020e927e96

```
--- FAIL: TestPlatformVerifier (2.85s)
    --- FAIL: TestPlatformVerifier/revoked_leaf (0.19s)
        root_darwin_test.go:116: unexpected verification error: got "x509: certificate has expired or is not yet valid: “revoked.badssl.com” certificate is expired", want "x509: “revoked.badssl.com” certificate is revoked"
    --- FAIL: TestPlatformVerifier/leaf_missing_SCTs (0.74s)
        root_darwin_test.go:118: unexpected verification success: want "x509: “no-sct.badssl.com” certificate is not standards compliant"
FAIL
FAIL	crypto/x509	4.436s
```

@rolandshoemaker any idea what is going on here? Do the builders need some certificate verification system dependency installed/enabled?

**Comment by bcmills** [comment]
FWIW, that's #57428.

**Comment by cagedmantis** [comment]
Just mentioning that this post-submit builder is failing in the 1.19 and 1.20 release branches.
